### PR TITLE
CEDS-1382 Add Remove Item Button

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/controllers/declaration/ItemsSummaryController.scala [hunk 1]
```diff
@@ -63,11 +63,22 @@ class ItemsSummaryController @Inject()(
       .map(_ => Redirect(controllers.declaration.routes.ProcedureCodesPageController.displayPage(newItem.id)))
   }
 
-  /*
-  Compilation warning
-    match may not be exhaustive.
-    [warn] It would fail on the following input: Left(_)
-   */
+  def removeItem(itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    exportsCacheService.get(journeySessionId) flatMap {
```
Comment (mattclark-zerogravit) [inline app/controllers/declaration/ItemsSummaryController.scala]: could we use the get item by id and sessionid here so we can do everything in one go?

Comment (ed0906) [reply app/controllers/declaration/ItemsSummaryController.scala]: Afraid not, that method just  returns the Export Item when I  need the whole cache object to be able to do the update

Comment (ed0906) [reply app/controllers/declaration/ItemsSummaryController.scala]: This kind of logic could potentially be pulled down into the service layer though I guess

Comment (mattclark-zerogravit) [reply app/controllers/declaration/ItemsSummaryController.scala]: yeah maybe. I suppose if you get the item.... then get the cachemodel to update its and extra db call. Your approach avoids that.

File: conf/declaration.routes [hunk 2]
```diff
@@ -117,7 +117,9 @@ POST        /nature-of-transaction               controllers.declaration.NatureO
 
 GET         /export-items                        controllers.declaration.ItemsSummaryController.displayPage()
 
-GET         /export-items/add                                  controllers.declaration.ItemsSummaryController.addItem()
+GET         /export-items/add                    controllers.declaration.ItemsSummaryController.addItem()
+
+GET         /export-items/:itemId/remove         controllers.declaration.ItemsSummaryController.removeItem(itemId)
```
Comment (MaciejRewera) [inline conf/declaration.routes]: IMHO it would be 'cleaner' to use DELETE in this place.

Comment (ed0906) [reply conf/declaration.routes]: Browser can only send POST and GET

**Pull request merged**